alibuild_helpers/build.py

```diff
@@ -1205,7 +1215,11 @@ def downloadTask(p):
 
     dieOnError(err, buildErrMsg)
 
-    syncHelper.syncToRemote(p, spec)
+    # Make sure not to upload local-only packages! These might have been
+    # produced in a previous run with a read-only remote store.
+    if not spec["revision"].startswith("local"):
```

Comment:
We should stop at the first local package, no?

Reply:
What should happen in the following case:
```
 D-1        E-local1
  |          |
 B-1        C-local1
   \        /
    A-local1
```
(i.e. A depends partly on local and partly on remote packages, but the remote packages are consistent; `buildOrder` is something like `[D, E, B, C, A]`)

I suppose `B-1` and `D-1` should've been uploaded in a previous alibuild run, so it might be alright to stop uploading when we hit `E-local1`. But if that previous upload failed, would we want it to happen now to make local and remote consistent again?

Reply:
I assume that if B-1 dependend on E, it would actually be called B-local1, no? If that is the case, you are right, we should not stop.

Reply:
Yes, if B depended on E, it would get a local revision number. When we have a read-only remote, no new packages with a non-local revision are produced, only existing ones reused.